transport/src/main/java/io/el/channel/local/LocalChannelRegistry.java

```diff
@@ -0,0 +1,29 @@
+package io.el.channel.local;
+
+import io.el.channel.Channel;
+import java.net.SocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class LocalChannelRegistry {
+
+  private static final ConcurrentMap<LocalAddress, Channel> boundChannels = new ConcurrentHashMap<>();
+
+  public static LocalAddress register(
+      Channel channel, LocalAddress oldLocalAddress, SocketAddress localAddress) {
```

Comment:
What is the role of the `oldLocalAddress`?

Reply:
it's to check if localAddress is already registered, but as you said I'd better to check outside of this registry class